### ISSUE TITLE
False positives for missing members

### ISSUE THREAD

**Issue by marmistrz** [PyCQA/pylint#1645]
### Steps to reproduce
Linted the following code: (from a unit test)
```
    def test_do_work(self, log, *_):
        self.client = Client(datadir=self.path, transaction_system=False,
                             connect_to_known_hosts=False,
                             use_docker_machine_manager=False,
                             use_monitor=False)

        c = self.client
        c.sync = Mock()
        c.p2pservice = Mock()
        c.task_server = Mock()
        c.resource_server = Mock()
        c.ranking = Mock()
        c.check_payments = Mock()

        # Test if method exits if p2pservice is not present
        c.p2pservice = None
        c.config_desc.send_pings = False
        # FIXME: Pylint doesn't handle mangled members well:
        # https://github.com/PyCQA/pylint/issues/1643
        c._Client__do_work()  # pylint: disable=no-member

        assert not log.exception.called
        assert not c.check_payments.called

        # Test calls with p2pservice
        c.p2pservice = Mock()
        c.p2pservice.peers = {
            str(uuid.uuid4()): Mock()
        }

        # FIXME: Pylint doesn't handle mangled members well:
        # https://github.com/PyCQA/pylint/issues/1643
        c._Client__do_work()  # pylint: disable=no-member

        assert not c.p2pservice.ping_peers.called
        assert not log.exception.called
        assert c.p2pservice.sync_network.called
        assert c.task_server.sync_network.called
        assert c.resource_server.sync_network.called
        assert c.ranking.sync_network.called
        assert c.check_payments.called

        # Enable pings
        c.config_desc.send_pings = True

        # Make methods throw exceptions
        def raise_exc():
            raise Exception('Test exception')

        c.p2pservice.sync_network = raise_exc
        c.task_server.sync_network = raise_exc
        c.resource_server.sync_network = raise_exc
        c.ranking.sync_network = raise_exc
        c.check_payments = raise_exc

        # FIXME: Pylint doesn't handle mangled members well:
        # https://github.com/PyCQA/pylint/issues/1643
        c._Client__do_work()  # pylint: disable=no-member

        assert c.p2pservice.ping_peers.called
        assert log.exception.call_count == 5
```

### Current behavior
```
E1101:Function 'raise_exc' has no 'called' member
```
at the following line
```
assert c.p2pservice.sync_network.called
```

### Expected behavior
No error, this is of course a `Mock`!

### pylint --version output
```
pylint 1.7.2, 
astroid 1.5.3
Python 3.6.2 (default, Jul 20 2017, 03:52:27) 
[GCC 7.1.1 20170630]
```

**Comment by PCManticore** [comment]
We don't plan to support mocks any time soon
